[Zacchaeus] iP

zacchaeuschok opened the pull request:
# Duke
> "Never misattribute a quote." - David Allen

Duke is the top chatbot for managing your tasks. It's

- fuss-free
- easy-to-learn
- text-based

All you need to do is:
1. download it from [here](https://github.com/zacchaeuschok/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :shipit:

And it is **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

zacchaeuschok committed:
Level-1

zacchaeuschok committed:
Level-2

zacchaeuschok committed:
Level-3

zacchaeuschok committed:
Level-4

zacchaeuschok committed:
Level-5

zacchaeuschok committed:
Level-6

zacchaeuschok committed:
A-Enums

zacchaeuschok committed:
fixed addToDo error checking

zacchaeuschok committed:
updated A-TextUiTesting

zacchaeuschok committed:
Add save file

zacchaeuschok committed:
add datetime recognition

zacchaeuschok committed:
Resolve merge conflicts with branch-Level-8

zacchaeuschok committed:
Modify save function

zacchaeuschok committed:
Extract parser

zacchaeuschok committed:
organise into duke package

zacchaeuschok committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

zacchaeuschok committed:
add JUnit tests

zacchaeuschok committed:
create JAR file

zacchaeuschok committed:
add javadocs

zacchaeuschok committed:
apply coding standard

zacchaeuschok committed:
add find command

zacchaeuschok committed:
resolve merge conflicts

ThomasHoooo commented on src/main/java/duke/parser/FileParser.java:
```diff
@@ -0,0 +1,82 @@
+
+import duke.exceptions.DukeException;
+import duke.exceptions.InvalidCommandException;
+import duke.tasks.*;
```
Don't import everything

ThomasHoooo commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,46 @@
+import java.time.LocalDateTime; // Import the LocalDateTime class
+import java.time.format.DateTimeParseException;
+
+public class Event extends Task{
```
Should have a space between Task and {

ThomasHoooo commented:
Looks good overall!

zacchaeuschok committed:
implement checkstyle

drkkjh commented on src/main/java/duke/exceptions/EmptyCommandException.java:
```diff
@@ -0,0 +1,14 @@
+package duke.exceptions;
+
+public class EmptyCommandException extends DukeException {
```
I like that you have used OOP here for specific DukeExceptions!

drkkjh commented on src/main/java/duke/parser/CommandParser.java:
```diff
@@ -0,0 +1,96 @@
+    public void handle(String command) {
+        try {
+            // Identify groups based on command pattern
+            Matcher matcher = COMMAND_PATTERN.matcher(command);
```
I like that you managed to use `enum` to handle the different cases 👍

drkkjh commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + TIME_PARSER.formatDeadline(this.deadline) + ")";
```
Great use of OOP here by calling constructor of superclass

drkkjh reviewed (commented):
Great use of OOP overall, LGTM.

zacchaeuschok committed:
add GUI

zacchaeuschok committed:
add fxml

zacchaeuschok committed:
create JAR file

zacchaeuschok committed:
add assertions

zacchaeuschok committed:
refactored code

zacchaeuschok committed:
Merge branch 'master' into A-Assertions

zacchaeuschok committed:
Merge pull request #1 from zacchaeuschok/A-Assertions

add assertions

zacchaeuschok committed:
use streams

zacchaeuschok committed:
Merge pull request #3 from zacchaeuschok/A-Streams

use streams

zacchaeuschok committed:
Merge pull request #3 from zacchaeuschok/A-Streams

zacchaeuschok committed:
use streams

Tasklist class: used streams to create filter function for tasklist

Previously there were common behaviours in printing tasklists for different methods e.g. find(), printDeadline().

The common behaviours across these methods caused code duplication.

Using a stream with filter function allows us to use predicates when dealing with subsets of tasklists.

zacchaeuschok committed:
Merge branch 'master' into A-CodeQuality

zacchaeuschok committed:
Merge pull request #4 from zacchaeuschok/A-CodeQuality

refactored code

zacchaeuschok committed:
detect duplicates

zacchaeuschok committed:
improve GUI

zacchaeuschok committed:
add personality

zacchaeuschok committed:
add GUI buttons

zacchaeuschok committed:
add more error handling

zacchaeuschok committed:
add joke function

zacchaeuschok committed:
add more tests

zacchaeuschok committed:
add more javadocs

zacchaeuschok committed:
set up CI

zacchaeuschok committed:
update gradle.yml file

zacchaeuschok committed:
Merge branch 'master' of https://github.com/zacchaeuschok/ip

zacchaeuschok committed:
set up CI

zacchaeuschok committed:
fix checkstyle

zacchaeuschok committed:
edit gradle properties

zacchaeuschok committed:
add user guide

zacchaeuschok committed:
update readme

zacchaeuschok committed:
Update help menu

zacchaeuschok committed:
Fix joke command

zacchaeuschok committed:
Fix checkstyle violations
